portable/ARMv8M/non_secure/port.c

```diff
@@ -67,6 +67,16 @@
 #if ( ( configRUN_FREERTOS_SECURE_ONLY == 1 ) && ( configENABLE_TRUSTZONE == 1 ) )
     #error TrustZone needs to be disabled in order to run FreeRTOS on the Secure Side.
 #endif
+
+/**
+ * Cortex-M23 does not have non-secure PSPLIM. We should use PSPLIM on Cortex-M23
+ * only when FreeRTOS runs on secure side.
+ */
+#if ( ( portHAS_ARMV8M_MAIN_EXTENSION == 0 ) && ( configRUN_FREERTOS_SECURE_ONLY == 0 ) )
```

Comment:
In a PE without the Main Extension but with the Security Extension, there are two stack limit registers in Secure
state for the purposes of stack limit checking. Therefore, we can still use PSPLIM register when `portHAS_ARMV8M_MAIN_EXTENSION` is 0 and `configRUN_FREERTOS_SECURE_ONLY` is 1. Therefore, this should be an `||` instead of an `&&`?

Reply:
Please update the comments above accordingly.

Reply:
There are following scenarios - 
1. Main extension is present ==> PSPLIM should be used.
2. Main extension is not present:
   a.  FreeRTOS runs on the secure side ==> PSPLIM should be used.
   b. FreeRTOS does not run on the secure side (i.e. runs on the non-secure side) ==> PSPLIM should not be used. 
   
So, the only scenario when PSPLIM should NOT be used is when - main extension is not present AND FreeRTOS does not run on the secure side. 

Therefore, I think it should be AND. Did I miss anything?